core/flyout_horizontal.js

```diff
@@ -245,16 +245,46 @@ Blockly.HorizontalFlyout.prototype.scrollToStart = function() {
 };
 
 /**
+ * Scroll the flyout to a position.
+ * @param {number} pos The targeted scroll position.
+ */
+Blockly.HorizontalFlyout.prototype.scrollTo = function(pos) {
+  this.scrollTarget = pos * this.workspace_.scale;
+
+  // Make sure not to set the scroll target past the farthest point we can
+  // scroll to, i.e. the content width minus the view width
+  var metrics = this.workspace_.getMetrics();
+  var contentWidth = metrics.contentWidth;
+  var viewWidth = metrics.viewWidth;
+  this.scrollTarget = Math.min(this.scrollTarget, contentWidth - viewWidth);
```

Comment:
Add `scrollTarget` to `Flyout`, initialized to `null` in the constructor, and add jsdoc saying what it is and what its possible values are (null or a number, as far as I can tell).

If it's only used in flyout and its children, mark it package.  I'd prefer private but I need to check if the closure compiler would throw warnings as a result.